backend/src/sms/services/sms-callback.service.ts

```diff
@@ -23,7 +24,14 @@ const isAuthenticated = (
   const [username, password] = credentials.split(':')
   const plainTextPassword =
     username + messageId + campaignId + config.get('smsCallback.callbackSecret')
-  return bcrypt.compareSync(plainTextPassword, password)
+
+  let isAuthenticated = getSha256Hash(plainTextPassword) === password
+  // if not passed with the new hash, retry with the old way
+  // TODO: remove this after all campaigns sent with the old way have completed
+  if (!isAuthenticated) {
+    isAuthenticated = bcrypt.compareSync(plainTextPassword, password)
```

Comment:
just curious: was under the impression that we should always use `bcrypt`'s method asynchronously, so as to prevent holding up the event loop

Reply:
This is inside the context of a request, which is already handled async